services/main.go

```diff
@@ -177,7 +181,19 @@ func (server *ApiService) fetchLatestVersion(params *omnitruck.RequestParams, c
 		data, err := server.DynamoServices(server.DatabaseService, c).VersionLatest(params)
 		if err != nil {
 			code, msg := getErrorCodeAndMsg(err)
-			server.logCtx(c).WithError(err).Error("Error while fetching the latest version for the " + params.Product)
+			server.logCtx(c).WithError(err).Error(utils.ErrorWhileFetchingLatestVersion + params.Product)
+			request.Failure(code, msg)
+			return data, &request
+		} else {
```

Comment:
No need for else. If you are returning in if

Reply:
Done